runtime_test.go

```diff
@@ -665,6 +665,49 @@ func TestRuntime_Close_ClosesCompiledModules(t *testing.T) {
 	}
 }
 
+func TestRuntime_Close_Instantiate_ShouldErr(t *testing.T) {
+	for _, tc := range []struct {
+		name            string
+		instantiateFunc func(r Runtime, mod CompiledModule) error
+	}{
+		{
+			name: "InstantiateModule",
+			instantiateFunc: func(r Runtime, mod CompiledModule) error {
+				_, err := r.InstantiateModule(testCtx, mod, NewModuleConfig())
+				return err
+			},
+		},
+		{
+			name: "CompileModule",
+			instantiateFunc: func(r Runtime, mod CompiledModule) error {
+				_, err := r.CompileModule(testCtx, binaryNamedZero)
+				return err
+			},
```

Comment:
shall we test other methods as well so that we never reach panics?

Reply:
InstantiateModule should be tested as that can be called without calling InstantiateModuleFromBinary. imho that's the last one

Reply:
it's now InstantiateModule + InstantiateModuleFromBinary + CompileModule so we should be good?